#163541092:  login user

Hoxtygen opened the pull request:
#### What does this PR do?
This route enables registered users to login to their account and
perform activities such as view their profile, vote, contest for
election.....etc. Because most of the routes are protected, a token is
generated if the user supplies a valid email and password, otherwise a
user is denied access into the application.

#### Description of Task to be completed?
Have the following endpoint working 
`POST /api/v1/auth/login`

#### How should this be manually tested?
clone the repo and cd into it then on the terminal do `npm run dev`
Using postman, test the endpoint with the following header information
- key: Content-Type value: application/json

#### What are the relevant pivotal tracker stories?
[Delivers [#163541092](https://www.pivotaltracker.com/story/show/163541092)]

#### Screenshots (if appropriate)

![loginuser](https://user-images.githubusercontent.com/19408669/51877310-0d69fa00-236c-11e9-95f0-8188a051107a.PNG)

Hoxtygen committed:
ft(feature): login user

This route enables registered users to login to their account and
perform activities such as view their profile, vote, contest for
election.....etc. Because most of the routes are protected, a token is
generated if the user supplies a valid email and password, otherwise a
user is denied access into the application.

[Delivers #163541092]

Orlayhemmy commented on server/controllers/userController.js:
```diff
@@ -11,9 +12,40 @@ class UserController {
   }
 
+  static login(req, res) {
+    req.checkBody('email', 'Email field is required').notEmpty().trim();
```
Extract this validation to another file. It is a good practice to have a validation file that is responsible for checking the users input before bringing it into the controller.

Hoxtygen replied:
Just got it done.

Hoxtygen committed:
refactor: extract input validation to another file

This refactor separates validation of inputs from the controller, it
makes sure that validation happens before data gets to the controller.

[Delivers #163541092]

Hoxtygen committed:
fix(typo):fix a typo in login validation file

This fix ensures email field is not empty

[Delivers [#163541092]
